### PR TITLE
Feature sb 74 public product page

### PR DESCRIPTION
**Pull request by ValentinaTenus**
1. added to product table product_variants to manage product quantity in specific size and color. Product variants includes size, color and quantity. Thats why I deleted color, size and quantity from product table.
2. added relations between products and users. every product includes user_id. also i specify one-to-many relation in user entity to be able to easily get all user's products.
3. according to above changes I changed findAll method in product repository and added findById
4. added changes in seeds
5. 
image

### CODE REVIEW

File: src/seeds/seed-data.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { Gender, ProductStatus } from '~/common/enums';
```
Comment (VladManukian) [inline src/seeds/seed-data.ts]: Why are you deleted alias?

File: .eslintrc.js [hunk 2]
```diff
@@ -11,6 +11,7 @@ module.exports = {
     'plugin:prettier/recommended',
     'airbnb-base',
     'airbnb-typescript/base',
+    "prettier"
```
Comment (VladManukian) [inline .eslintrc.js:14]: Why do you add this? We have alredy `plugin:prettier/recommende`. It is work well.

File: src/seeds/combined-seeder.ts [hunk 3]
```diff
@@ -86,63 +91,83 @@ export class CombinedSeeder implements Seeder {
...
-      const size = dataSource
-        .getRepository(SizeEntity)
-        .findOneBy({ id: product.size_id });
+      const { variants, image_ids: imageIds, ...productData } = product;
+
+      async function fetchImageById(
+        imageId: number,
+      ): Promise<ImageEntity | null> {
+        return imageRepository.findOne({ where: { id: imageId } });
+      }
```
Comment (Vfivew) [inline src/seeds/combined-seeder.ts:114]:
```suggestion
    const fetchImageById = async (imageId: number): Promise<ImageEntity | null> =>
  imageRepository.findOne({ where: { id: imageId } });

```

File: .eslintrc.js (hunk 2 above)
Comment (ValentinaTenus) [reply .eslintrc.js:14]: to resolve ESLint rules and Prettier conflict. with only 'plugin:prettier/recommended', it does't work.

**Review by VladManukian — APPROVED** [review]

**Review by likeri29 — APPROVED** [review]

**Review by jaroslawtt — APPROVED** [review]

**Comment by jaroslawtt** [comment]
conflicts

**Pull request merged**